### PR TITLE
Dead code, switch cleanup and `crypto` bump.

### PR DESCRIPTION
**Pull request by pq**
Prep to get us ready to pull in a new dev SDK and bump our analyzer DEP.
- updates `crypto` (required by fresh analyzer)
- fixes newly flagged dead code warnings
- fixes switches that fall through and don't return

cc @abarth, @Hixie

### CODE REVIEW

File: packages/flutter/lib/src/rendering/block.dart [hunk 1]
```diff
@@ -187,6 +189,8 @@ class RenderBlock extends RenderBox
       case Axis.vertical:
         return _getIntrinsicCrossAxis((RenderBox child) => child.getMinIntrinsicWidth(height));
     }
+    assert(mainAxis != null);
```
Comment (abarth) [inline packages/flutter/lib/src/rendering/block.dart]: These should probably just be `assert(false)`

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/rendering/block.dart (hunk 1 above)
Comment (pq) [inline packages/flutter/lib/src/rendering/block.dart]: @Hixie and I chatted about these cases and his suggestion (I think) was to assert not `null`.  Happy to do whatever you guys agree on though! 👍

Comment (abarth) [inline packages/flutter/lib/src/rendering/block.dart]: Ok, that's fine with me too.

Comment (Hixie) [inline packages/flutter/lib/src/rendering/block.dart]: The way you have it is fine IMHO. I think I would have put the conditional assert above the switch or the false assert below the switch but it really is neither here nor there and, again, what you have is totally fine IMHO.

File: packages/flutter/pubspec.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ dependencies:
   vector_math: '>=2.0.3 <3.0.0'
 
   # We need to pin crypto because archive can't handle larger numbers.
-  crypto: 0.9.2
+  crypto: '>=1.1.1 <3.0.0'
```
Comment (Hixie) [inline packages/flutter/pubspec.yaml]: we're not pinning crypto any more here so the comment is wrong.

Comment (pq) [inline packages/flutter/pubspec.yaml]: Right! Meant to fix that.  Thanks.

**Pull request merged**